src/main/java/org/broadinstitute/hellbender/engine/DuplicateSetWalker.java

```diff
@@ -0,0 +1,116 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMTag;
+import org.apache.commons.lang3.tuple.Pair;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.walkers.consensus.MoleculeID;
+import org.broadinstitute.hellbender.tools.walkers.consensus.ReadsWithSameUMI;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * A walker that processes duplicate reads that share the same Unique molecule Identifier (UMI) as a single unit.
+ *
+ * This tool assumes that the input bam has been sorted by UMI (the {@link SAMTag.MI} tag to be specific) with FGBio GroupReadsByUmi:
+ * http://fulcrumgenomics.github.io/fgbio/tools/latest/GroupReadsByUmi.html
+ */
+public abstract class DuplicateSetWalker extends ReadWalker {
+    public static final String MIN_REQUIRED_READS_NAME = "min-reads";
+    public static final String MIN_REQUIRED_READS_PER_STRAND_NAME = "min-per-strand-reads";
+
+    private static final int DEFAULT_MINIMUM_READS_PER_SET = 1;
+    private static final int DEFAULT_MINIMUM_READS_PER_STRAND = 0;
+
+    @Argument(fullName = MIN_REQUIRED_READS_NAME, doc = "The mininum total number of reads required in the set", optional = true)
+    private int minimumRequiredReadsPerUMI = DEFAULT_MINIMUM_READS_PER_SET;
+
+    // The user may choose to keep read sets containing both strands (i.d. duplex evidence) only by setting this argument to a positive number
```

Comment:
Wording.  This would be clearer as "The user may choose to only keep read sets containing both strands by setting this argument to a positive number."  

I misread this at first as the "read sets containing both strands will be kept ONLY if this is set to a positive number"

Reply:
Ah I went back and forth about this and I clearly chose wrong. done.